File: check-if-it-is-a-good-array.py
def gcd(a, b):
    while b:
        a, b = b, a % b
    return a

class Solution:
    def isGoodArray(self, nums: List[int]) -> bool:
        res = nums[0]
        for el in nums:
            res = gcd(res, el)
        return res == 1

File: check-if-it-is-a-good-array.cpp
class Solution {
public:
    int gcd(int a, int b) {
        while (b) {
            int temp = a;
            a = b;
            b = temp % a;
        }
        return a;
    }

    bool isGoodArray(vector<int>& nums) {
        int g = 0;
        for (auto el : nums) g = gcd(g, el);
        return g == 1;
    }
};